plugin/sampling/strategystore/static/strategy_store.go

```diff
@@ -83,32 +84,73 @@ func (h *strategyStore) Close() {
 	h.cancelFunc()
 }
 
-func (h *strategyStore) autoUpdateStrategies(interval time.Duration, filePath string) {
+func (h *strategyStore) autoUpdateStrategies(interval time.Duration, opts Options) {
+	// Default reload option.
+	reloadFrom := "file"
+	path := opts.StrategiesFile
+	if opts.StrategiesURL != "" {
+		reloadFrom = "url"
+		path = opts.StrategiesURL
+	}
+	
 	lastValue := ""
 	ticker := time.NewTicker(interval)
 	defer ticker.Stop()
 	for {
 		select {
 		case <-ticker.C:
-			lastValue = h.reloadSamplingStrategyFile(filePath, lastValue)
+			lastValue = h.reloadSamplingStrategy(reloadFrom, path, lastValue)
 		case <-h.ctx.Done():
 			return
 		}
 	}
 }
 
-func (h *strategyStore) reloadSamplingStrategyFile(filePath string, lastValue string) string {
-	currBytes, err := ioutil.ReadFile(filepath.Clean(filePath))
-	if err != nil {
-		h.logger.Error("failed to load sampling strategies", zap.String("file", filePath), zap.Error(err))
-		return lastValue
+func (h *strategyStore) reloadSamplingStrategy(reloadFrom string, path string, lastValue string) string {
```

Comment:
The caller of this function already does checks for file vs. url. I would pass a loader function here that abstracts the source of the new config, e.g. `loader func () (string, error)`, and only deal with processing the new value.

Reply:
Yes, that's better. Was going to do the same but changed my mind later.